source/actions/index.js

```diff
@@ -12,9 +12,9 @@ const requestRoute = (index) => ({
   payload: { index }
 })
 
-const receiveRouteSuccess = (index, points) => ({
+const receiveRouteSuccess = (index, { points, distance }) => ({
```

Comment:
Correct. For some reason, I thought the distance in each of those points was the distance of that particular segment, rather than the distance from the start. I shall remove.